vertx-mutiny-clients/vertx-mutiny-sql-client/src/main/java/io/vertx/mutiny/sqlclient/SqlClientHelper.java

```diff
@@ -21,28 +22,22 @@
      */
     public static <T> Multi<T> inTransactionMulti(Pool pool, Function<SqlClient, Multi<T>> sourceSupplier) {
         return usingConnectionMulti(pool, conn -> {
-            return conn
-                    .begin()
-                    .onItem().transformToMulti(transaction -> {
-                        Multi<T> multi = sourceSupplier.apply(conn)
-                                .onCompletion().call(() -> transaction.commit())
-                                .onFailure().call(() -> transaction.rollback());
-                        return multi;
-                        //                        return multi.onTermination()
-                        //                                .call((failure, cancellation) -> {
-                        //                                    System.out.println("Failing or cancelling " + failure);
-                        //                                    if (cancellation || failure != null) {
-                        //                                        // The failure will be propagated downstream
-                        //                                        System.out.println("rollback");
-                        //                                        return transaction.rollback();
-                        //                                    } else {
-                        //                                        return transaction.commit();
-                        //                                    }
-                        //                                });
-                    });
+            return conn.begin().onItem().transformToMulti(tx -> {
+                Multi<T> multi = Multi.createBy().concatenating().streams(
```

Comment:
Also I'm not sure why you concatenate here

Reply:
Concatenating allows to:

1. Make sure the the commit stream is subscribed only after data items have been emitted without any error
2. Return a multi that, when subscribed, emits all the data items.

I got to a similar implementation originally in `vertx-rx`. I don't believe that RxJava has an operator on `Flowable` that allows to execute an action on termination and complete/fail the `Flowable` stream depending on the action result? Can Mutiny do that? I mean, other than by concatenating two `Multi`, of course :-)

Reply:
Mutiny can do this, this is done using the `call` method.